content/docs/nextjs/github-public-repo.md

```diff
@@ -0,0 +1,292 @@
+
+## Auth Redirects
+
+We will also need a page to redirect the user to while authenticating with GitHub. This component will render as a modal during authentication.
+
+```tsx
+// pages/github/authorizing.tsx
+
+import { useGithubAuthRedirect } from 'react-tinacms-github'
+
+// Our GitHub app redirects back to this page with auth code
+export default function Authorizing() {
+  // Let the main app know, that we received an auth code from the GitHub redirect
+  useGithubAuthRedirect()
+
+  return <h2>Authorizing with GitHub, Please wait...</h2>
+}
+```
+
+## Setup the GitHub OAuth App
+
+In GitHub, within your Account Settings, click <a href="https://github.com/settings/developers" target="_blank">Oauth Apps</a> under Developer Settings. Go ahead and create a "New Oauth App".
+
+For the **Authorization callback URL**, enter the url for the "/github/authorizing" page that you created above (e.g https://your-url/github/authorizing). Fill out the other fields with your custom values.
+
+> _Note:_ If you are **testing your app locally**, you may need a separate development GitHub app (with a localhost redirect), and a production GitHub app.
+
+Next we will add the **Client ID** and **Client Secret** to your environment variables.
+
+<!-- TODO: add ## Developing Locally section at this point? to explain where the content is sourcing from, etc. -->
+
+### Setting Environment Variables
+
+This guide will reference [environment variables](https://nextjs.org/docs/api-reference/next.config.js/environment-variables) — these are sensitive values specific to your project. The Tina helpers will use these value to talk to your repository, enabling auth and data fetching via GitHub.
+
+To set these variables, create a `.env` file in your project root:
+
+```
+# .env
+GITHUB_CLIENT_ID= # Taken from GitHub
+GITHUB_CLIENT_SECRET= # Taken from Github
+REPO_FULL_NAME=tinacms/tinacms.org # This is your github repository's owner / repo-name.
+BASE_BRANCH=master
+```
+
+You can use the `dotenv` package to load the `.env` file:
+
+```bash
+$ npm install --save dotenv
+```
+
+Now, to load these `.env` values in the front-end, your [next.config](https://nextjs.org/docs/api-reference/next.config.js/introduction) file will need to be configured.
+
+```js
+// next.config.js
+
+require('dotenv').config()
+
+module.exports = {
+  env: {
+    GITHUB_CLIENT_ID: process.env.GITHUB_CLIENT_ID,
+    REPO_FULL_NAME: process.env.REPO_FULL_NAME,
+    BASE_BRANCH: process.env.BASE_BRANCH,
+  },
+  // ...
+}
+```
+
+> Note that we did not add `GITHUB_CLIENT_SECRET` to the config exports. This is the a secret key that should only be used from the server and should not be accessible through the browser.
+
+## Loading content from GitHub
+
+Now that we have access to the user's auth token, we can load content from GitHub within [_getStaticProps_](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation).
```

Comment:
We should mention the `../content/home.json`.

If that file doesn't exist they could point it to an existing file.

If they don't have any files yet they should create `content/home.json` and make sure to push it to GItHub